tests/conftest.py

```diff
@@ -1,8 +1,78 @@
+import json
 import os
+import tempfile
 import pytest
 
+from datetime import datetime
+from securedrop_client.config import Config
+from securedrop_client.models import Base, make_engine, Source
+from sqlalchemy.orm import sessionmaker
+from uuid import uuid4
+
+
+with open(os.path.join(os.path.dirname(__file__), 'files', 'test-key.gpg.pub.asc')) as f:
+    PUB_KEY = f.read()
+
+
+@pytest.fixture(scope='function')
+def homedir():
+    '''
+    Create a "homedir" for a client.
+
+    Using `mkdtemp` and not `TemporaryDirectory` because the latter will remove the directory
+    when the object is destroyed, and we want to leave it on the file system so developers can
+    inspect the contents for debugging purposes.
+    '''
+
+    tmpdir = tempfile.mkdtemp(prefix='sdc-')
+    os.chmod(tmpdir, 0o0700)
+
+    data_dir = os.path.join(tmpdir, 'data')
+    gpg_dir = os.path.join(tmpdir, 'gpg')
+    logs_dir = os.path.join(tmpdir, 'logs')
+
+    for dir_ in [data_dir, gpg_dir, logs_dir]:
+        os.mkdir(dir_)
+        os.chmod(dir_, 0o0700)
+
+    yield tmpdir
```

Comment:
Right but the `TemporaryDirectory` deletes these when the context manager disappears or when the object gets GC'd, so they would never be able to look at the files unless they jumped in while the test was running.

We could do something pytest-like where we reuse the test name in the directory name so they aren't recreated per-test, but at that point we're just copying the behavior of `tmpdir` anyway. I'm not sure what the way around this is. Maybe just use a simple counter so that we only ever have `N` dirs, where `N` is the number of times `homedir` was used during a single `pytest` session?

Reply:
ok in the spirit of not reimplementing `tmpdir`, let's leave this as is for now ;)